### ISSUE TITLE
coverage report produces too many Anys

### ISSUE THREAD

**Issue by terencehonles** [python/mypy#10325]
**Bug Report**

When generating a coverage report there are too many `Any`s reported for lines. I've figured out how it's counting `Any`s and I may be able to help with a patch.

**To Reproduce**

1. ```python
   def test() -> str:
     value: str = getattr(test, '__name__')
     return value
   ```
2. `mypy --html-report . test.py`
3. View generated `html/test.py.html`

**Expected Behavior**

As mypy reported a success, I would expect no `Any`s, but maximally I would expect 1.

**Actual Behavior**

Mypy reports 4 explicit anys on the `getattr` line.

When stepping through the `StatisticsVisitor` visit I noticed this is because when visiting a call expression:

https://github.com/python/mypy/blob/d089891198ef470c8bec9bd7d7b50a02757c5b68/mypy/stats.py#L217-L225

1. The call expression itself is processed, which first starts with the node itself. Because the result of `getattr` is `Any` the visitor has `Any` in its `typemap` for that node. This is arguably the *only* `Any` that I would expect, and it probably should still not be reported because there is an explicit type on the assignment (I do note that assignment *could* be wrong, and maybe this should be reported as a partial cover).
2. The callee is then visited which will be processing the node with type `def (Any, name: builtins.str, Any =) -> Any`, in `StatisticsVisitor.type` it calls `get_proper_types(collect_all_inner_types(t)) + [t]` which is `[Any, builtins.str, Any, Any, def (Any, name: builtins.str, Any =) -> Any]` which will add 3 more `Any`s to the line. Now that I'm looking at it closer, if I had added `Any`s to the type definition of `test` they would increase the number of `Any`s on the line when visiting the arguments to `getattr`. 

**Your Environment**

- Mypy version used: `mypy 0.812`
- Mypy command-line flags: `--html-report .`
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: `Python 3.8.8`
- Operating system and version: N/A
